embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/compression/CompressionOutputStream.kt

```diff
@@ -0,0 +1,70 @@
+package io.embrace.android.embracesdk.internal.compression
+
+import java.io.OutputStream
+import java.util.zip.GZIPOutputStream
+
+/**
+ * This Output Stream is used to compress the data that is being written to it.
+ * It will only compress the data if it's not already compressed.
+ * Given we started compressing payloads to disk from version 6.3.0 onwards,
```

Comment:
This is a standalone component, so a description of why we are using it doesn't seem appropriate. Maybe adding this a the place we are using it is better?